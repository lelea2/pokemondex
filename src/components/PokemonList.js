import React, { Component } from 'react';
import PokemonCell from './PokemonCell';
import PropTypes from 'prop-types';
import './styles/PokemonList.css';

class PokemonList extends Component {

  renderList() {
    const { data, pokemonDetail } = this.props;
    const currentName = (pokemonDetail) ? (pokemonDetail.name) : null;
    return data.map((item, i) => {
      return <PokemonCell
        key={`${i}${item}`}
        {...item}
        getDetail={this.props.getDetail}
        active={(item.name === currentName) ? true : false}
      />
    });
  }

  render() {
    return (
      <section className="poke-list">
        {this.renderList()}
      </section>
    );
  }
}

PokemonList.propTypes = {
  data: PropTypes.array,
  pokemonDetail: PropTypes.object,
  getDetail: PropTypes.func
};

export default PokemonList;
