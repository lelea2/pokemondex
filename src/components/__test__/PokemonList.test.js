import React from 'react';
import renderer from 'react-test-renderer';
import PokemonList from '../PokemonList';

describe('Component: PokemonList', () => {
  it("PokemonList - with active pokemondetail renders correctly", () => {
    const pokemonDetail = { name: 'venusaur' };
    const data = [{
      name: 'Pikachu',
      url: ''
    }, {
      name: 'venusaur',
      url: ''
    }]
    const test = renderer.create(<PokemonList data={data} pokemonDetail={pokemonDetail} getDetail={() => null} />).toJSON();
    expect(test).toMatchSnapshot();
  });

  it("PokemonList - default renders correctly", () => {
    const data = [{
      name: 'Pikachu',
      url: ''
    }, {
      name: 'venusaur',
      url: ''
    }]
    const test = renderer.create(<PokemonList data={data} pokemonDetail={null} getDetail={() => null} />).toJSON();
    expect(test).toMatchSnapshot();
  });
});