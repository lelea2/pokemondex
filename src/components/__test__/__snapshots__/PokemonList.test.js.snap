// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonList PokemonList - default renders correctly 1`] = `
<section
  className="poke-list"
>
  <button
    className="poke-cell"
    onClick={[Function]}
  >
    <img
      alt="Pikachu"
      height="50"
      src="https://img.pokemondb.net/artwork/Pikachu.jpg"
      width="50"
    />
  </button>
  <button
    className="poke-cell"
    onClick={[Function]}
  >
    <img
      alt="venusaur"
      height="50"
      src="https://img.pokemondb.net/artwork/venusaur.jpg"
      width="50"
    />
  </button>
</section>
`;

exports[`Component: PokemonList PokemonList - with active pokemondetail renders correctly 1`] = `
<section
  className="poke-list"
>
  <button
    className="poke-cell"
    onClick={[Function]}
  >
    <img
      alt="Pikachu"
      height="50"
      src="https://img.pokemondb.net/artwork/Pikachu.jpg"
      width="50"
    />
  </button>
  <button
    className="poke-cell active"
    onClick={[Function]}
  >
    <img
      alt="venusaur"
      height="50"
      src="https://img.pokemondb.net/artwork/venusaur.jpg"
      width="50"
    />
  </button>
</section>
`;
